### PR TITLE
Feature/calendar templates

### PR DESCRIPTION
**Pull request by freekvandeven**
This one needs to be merged second

### CODE REVIEW

File: packages/flutter_availability/lib/src/ui/widgets/template_legend.dart [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            ),
+          ],
+        ),
+      ),
+    );
+    return Column(
+      children: [
+        GestureDetector(
+          onTap: () {
+            if (!templatesAvailable) {
```
Comment (PureTryOut) [inline packages/flutter_availability/lib/src/ui/widgets/template_legend.dart]: Extract functions outside of the widget tree

File: packages/flutter_availability/lib/src/ui/widgets/template_legend.dart [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.center,
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Text(
+                  translations.templateLegendTitle,
+                  style: textTheme.titleMedium,
+                ),
+                // a button to open a drawer with all the templates
+                if (templatesAvailable)
```
Comment (PureTryOut) [inline packages/flutter_availability/lib/src/ui/widgets/template_legend.dart]: Add brackets.

File: packages/flutter_availability/lib/src/ui/widgets/template_legend.dart [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                color: theme.colorScheme.onSurface,
+                width: 1,
+              ),
+            ),
+            padding: const EdgeInsets.only(right: 2),
+            child: Column(
+              children: [
+                Container(
+                  constraints: const BoxConstraints(
+                    maxHeight: 100,
```
Comment (PureTryOut) [inline packages/flutter_availability/lib/src/ui/widgets/template_legend.dart]: Remove the trailing comma

File: packages/flutter_availability/lib/src/ui/widgets/template_legend.dart [hunk 4]
```diff
@@ -87,57 +88,71 @@ class _TemplateLegendState extends State<TemplateLegend> {
...
+                              );
+                            },
+                          ),
+                        ),
+                      ),
+                    ],
                   ),
+                )
+              : const Divider(
+                  height: 1,
```
Comment (PureTryOut) [inline packages/flutter_availability/lib/src/ui/widgets/template_legend.dart]: Remove trailing comma

**Review by PureTryOut — CHANGES_REQUESTED** [review]

**Comment by freekvandeven** [comment]
I added brackets around the if statement to make you happy @PureTryOut  Even tough it is not required by our coding standard

**Review by PureTryOut — APPROVED** [review]

**Pull request merged**